src/auth/service/Auth.service.ts

```diff
@@ -0,0 +1,106 @@
+
+        @Inject('AUTH0_CLIENT_ID')
+        private readonly AUTH0_CLIENT_ID: string,
+
+        @Inject('AUTH0_CLIENT_SECRET')
+        private readonly AUTH0_CLIENT_SECRET: string,
+
+        @Inject('AUTH0_AUTHORIZE_ROUTE')
+        private readonly AUTH0_AUTHORIZE_ROUTE: string,
+
+        @Inject('AUTH0_GET_TOKEN_ROUTE')
+        private readonly AUTH0_GET_TOKEN_ROUTE: string,
+
+        @Inject('AUTH0_AUTH_API_AUDIENCE')
+        private readonly AUTH0_AUTH_API_AUDIENCE: string,
+    ) {
+        this.validateIfExisting([
+            this.BASE_URL,
+            this.AUTH0_BASE_URL,
+            this.AUTH0_CLIENT_ID,
+            this.AUTH0_CLIENT_SECRET,
+            this.AUTH0_AUTHORIZE_ROUTE,
+            this.AUTH0_GET_TOKEN_ROUTE,
+            this.AUTH0_AUTH_API_AUDIENCE,
+        ]);
+    }
+
+    public getAuthUrl(): string {
+        const queryParams: URLSearchParams = new URLSearchParams({
+            audience: this.AUTH0_AUTH_API_AUDIENCE,
+            response_type: 'code',
+            client_id: this.AUTH0_CLIENT_ID,
+            redirect_uri: `${this.BASE_URL}${Routes.AUTH_ROUTE}${Routes.AUTH_CALLBACK_ROUTE}`,
+            scope: 'offline_access',
+        });
+        return `${this.AUTH0_BASE_URL}${this.AUTH0_AUTHORIZE_ROUTE}?${queryParams}`;
+    }
+
+    public async getToken(code: string): Promise<string> {
+        try {
+            const response: AxiosResponse = await axios({
+                method: 'POST',
+                url: `${this.AUTH0_BASE_URL}${this.AUTH0_GET_TOKEN_ROUTE}`,
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                },
+                data: {
+                    grant_type: 'authorization_code',
+                    client_id: this.AUTH0_CLIENT_ID,
+                    client_secret: this.AUTH0_CLIENT_SECRET,
+                    code,
+                    redirect_uri: `${this.BASE_URL}${Routes.AUTH_ROUTE}${Routes.AUTH_CALLBACK_ROUTE}`,
+                },
+            });
+            return response.data.access_token;
+        } catch (err: any) {
+            throw new AuthException('Error while getting an auth token', {
+                cause: err,
+            });
+        }
+    }
+
+    public async refreshToken(refreshToken: string): Promise<void> {
+        try {
+            const response: AxiosResponse = await axios({
+                method: 'POST',
+                url: `${this.AUTH0_BASE_URL}${this.AUTH0_GET_TOKEN_ROUTE}`,
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                },
+                data: {
+                    grant_type: 'refresh_token',
+                    client_id: this.AUTH0_CLIENT_ID,
+                    client_secret: this.AUTH0_CLIENT_SECRET,
+                    refresh_token: refreshToken,
+                },
+            });
+            console.log('data ----> ', response.data); // to implement
```

Comment:
?

Reply:
to właśnie moja wątpliwość, bo w sumie to jak już wdrożyłem ten flow to się zorientowałem, że skoro Auth0 jest SSO to samo pozyskanie i potem odświeżanie tokena powinno być implementowane na froncie (np Iron session w Next czy Passport)  a w naszej apce powinniśmy tylko mieć guardy na te tokeny (co też już wdrożyłem i działa) oraz autoryzacje ról (właśnie wdrażam). Wtedy wyleci w ogóle możliwość pozyskiwania tokenów z naszej apki bo przekazujemy je tylko z Auth0 to usera a może sobie pozyskać token sam. Refresh token tym bardziej do wywalenia. Wywaliłļym to teraz ale pozyskiwanie tokenów na razie pozostawił, żebyśmy mogli w ogóle testować tą apkę i nie musięli odpalać drugiego środowiska czy robić jakiegoś frontu na prędce, gdzie byłoby pozyskiwanie tokenów. W razie gdyby, niejasno nappisał, możemy się zdzwonić na 15 min :)